flocker/control/_protocol.py

```diff
@@ -355,7 +380,25 @@ def set_node_era(self, era, node_uuid):
         return {}
 
 
-class ControlAMP(AMP):
+class FlockerServiceAMP(AMP):
+    """
+    Base AMP protocol that provides a method to forcefully close the
+    transport connection.
+    """
+    def __init__(self, reactor, *args, **kwargs):
+        self._timeout = Timeout(
+            reactor, 2 * PING_INTERVAL.seconds, self.close_connection)
+        AMP.__init__(self, *args, **kwargs)
+        locator = kwargs.pop('locator', None)
+        if locator:
+            locator.timeout_reset = self._timeout.reset
```

Comment:
`self._timeout` doesn't get used anywhere else, so it doesn't need to be an attribute.